src/clients/python/experimental_api_v2/examples/grpc_v2_image_client.py

```diff
@@ -237,23 +239,32 @@ def requestGenerator(input_name, output_name, c, h, w, format, dtype, FLAGS,
         image_data.append(preprocess(img, format, dtype, c, h, w,
                                      FLAGS.scaling))
 
-    # Send requests of FLAGS.batch_size images.
-    input_bytes = None
-    for idx in range(FLAGS.batch_size):
-        # wrap over if requested batch size exceeds number of provided images
-        img_idx = idx % len(filenames)
-        input_filenames.append(filenames[img_idx])
-        if input_bytes is None:
-            input_bytes = image_data[img_idx].tobytes()
-        else:
-            input_bytes += image_data[img_idx].tobytes()
-
-    input_contents = grpc_service_v2_pb2.InferTensorContents()
-    input_contents.raw_contents = input_bytes
-    input.contents.CopyFrom(input_contents)
-    request.inputs.extend([input])
-
-    yield request
+    # Send requests of FLAGS.batch_size images. If the number of
+    # images isn't an exact multiple of FLAGS.batch_size then just
+    # start over with the first images until the batch is filled.
+    image_idx = 0
+    last_request = False
+    while not last_request:
```

Comment:
If the number of images is not equal to batch size shouldn't the images be repeated as described in the comments?
The check should remain `for idx in range(FLAGS.batch_size):`
inside the loop having `image_idx = image_idx % len(filenames)` retains this behavior. 

Additionally why do you need a `result_filenames`? The name here refers to the name of the input file.

Reply:
The point of having `for idx in range(FLAGS.batch_size):` is to ensure the inner loop runs for `batch_size` times. The `idx` must not be confused with `image_idx`. The `image_idx` will loop around the available images to create the inputs.
For instance, if the batch_size is 2 and the num of images in the directory are 3(say A, B and C), then the logic selects A and B for the first batch and C and A for the second batch. Hence, starting over the first image as described in the comment.

result_filenames is the list of list of input filenames. From the above example, it will be [[A,B], [C,A]]. The client needs to record the filename to print along with the inferred label.